### PR TITLE
Fix Up arrow and most F keys, etc.

### PR DESCRIPTION
**Pull request by suresttexas00**
Per my other message, https://github.com/magmax/python-readchar/issues/20#issuecomment-286588396

#1 way was the best I can manage for now!

This works (but I am not sure I need to have the `\x4f\ now since that is not actually part of the F key codes. (at least not on linux?) 

 EDIT: removed the `\x4f\` checking too..
2ND EDIT: re-added the `\x4f\` for F1-F4.

### CODE REVIEW

File: readchar/key.py [hunk 1]
```diff
@@ -35,23 +46,29 @@
...
 HOME = '\x1b\x5b\x48'
 END = '\x1b\x5b\x46'
 
 INSERT = '\x1b\x5b\x32\x7e'
-SUPR = '\x1b\x5b\x33\x7e'
 
+# what's this?  Why is it defined to '' in above Line 11?
+# why is it the same as DELETE?
+SUPR = '\x1b\x5b\x33\x7e'
+DELETE = '\x1b\x5b\x33\x7e'
```
Comment (magmax) [inline readchar/key.py]: I'm affraid it is wrong.
It should be:

```
SUPR = '\x1b\x5b\x33\x7e'
DELETE='0x7f'
BACKSPACE=DELETE
```

File: readchar/__init__.py [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# -*- coding: utf-8 -*-
+# This file is based on this gist:
+# http://code.activestate.com/recipes/134892/
+# So real authors are DannyYoo and company.
```
Comment (magmax) [inline readchar/__init__.py]: Not really.
This file is already mine :)
The pieces taken from that gist are the ones under readchar_*, and they have been modified a lot, but I like to maintain the authority.
Anyways, the comment is welcome here too.

**Comment by suresttexas00** [comment]
When I enable the the print statement in readchar_linux.py;
I get this for Delete: 
```
('\x1b', '0x1b')
('[', '0x5b')
('3', '0x33')
('~', '0x7e')
```

I get this for Backspace:
```
('\x7f', '0x7f')
```

I don't have a key labeled "SUPR", so I don't know what it is.

**Comment by suresttexas00** [comment]
I created this little interactive tester for use in reading what keys produce:
```
# -*- coding: utf-8 -*-

import readchar

while True:

    x = readchar.readkey()
    if x == readchar.key.F5:
        print("terminated.")
        break
    print("key read len was: %s" % len(x))
    if x == readchar.key.INSERT:
        print(" key")

```

**Comment by suresttexas00** [comment]
I also removed the Danny Yoo attribution in the  Key.py file.

File: readchar/__init__.py (hunk 2 above)
Comment (suresttexas00) [inline readchar/__init__.py]: Removed it.

**Comment by suresttexas00** [comment]
This is correct because on English keyboards there is no such key as SUPR ... our keyboards it is DELETE and is not the same as BackSPACE...
https://www.pcreview.co.uk/threads/spanish-keyboard.2840896/

**Comment by suresttexas00** [comment]
My only concern in all this is does it still work in windows.  Does the travis test for windows?

File: readchar/readchar.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def readkey(getchar_fn=None):
     if ord(c1) != 0x1b:
         return c1
     c2 = getchar()
-    if ord(c2) != 0x5b:
+    if ord(c2) not in (0x5b, 0x4f):
```
Comment (suresttexas00) [inline readchar/readchar.py]: This was required after all to Handle the F1-4 keys.

Comment (suresttexas00) [inline readchar/readchar.py]: Of course, we still can''t handle F 11, 12 because of the same concept as the ESC key problem... :(

**Comment by suresttexas00** [comment]
I am going to close this and do a re-pull for clarity.  I have a new solution that also works for all Linux keys, including the pesky ESC key.  I also implements a version of @guiweber windows code, although it tries to use the Linux type constants in our key.py file versus the string Windows names.

**Pull request closed without merging**